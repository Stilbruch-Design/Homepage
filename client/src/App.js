import React from "react";
import { BrowserRouter as Router, Route, Switch, Link } from "react-router-dom";
import { Provider } from "react-redux";
import CookieConsent from "react-cookie-consent";
import store from "./store";

import Navbar from "./components/Navbar";
import Footer from "./components/Footer";
import Main from "./components/Main";
import Privacy from "./components/Privacy";
import Terms from "./components/Terms";

export default function App() {
  return (
    <Provider store={store}>
      <Router>
        <div>
          <Navbar />
          <Switch>
            <Route exact path="/" component={Main} />
            <Route exact path="/datenschutz" component={Privacy} />
            <Route exact path="/impressum" component={Terms} />
            <Route path="/*" component={Main} />
          </Switch>
          <hr />
          <Footer />
          <CookieConsent
            acceptOnScroll
            acceptOnScrollPercentage={70}
            location="bottom"
            buttonText="Ok, stimme zu."
            cookieName="stilbruch.dev_cookie"
            debug="true"
            style={{
              alignItems: "center",
              display: "flex",
              flexDirection: "row",
              justifyContent: "space-around",
              borderRadius: "10px",
              boxShadow: "5px 10px 20px rgba(0,0,0, 0.8)",
              background: "rgba(24, 24, 24, 0.9)",
              textAlign: "center",
              color: "#FF0000",
              textDecoration: "none",
              fontSize: "2rem",
              fontWeight: "bold",
              marginBottom: "1rem"
            }}
            buttonStyle={{
              border: "1px solid #FF0000",
              borderRadius: "10px",
              boxShadow: "5px 10px 20px rgba(0,0,0, 0.5)",
              background: "rgba(95, 95, 95, 0.3)",
              color: "rgb(255, 0, 0)",
              fontSize: "2rem",
              fontFamily: "inherit",
              display: "flex",
              flexDirection: "column",
              justifyContent: "center",
              outline: "none",
              fontWeight: "bold"
            }}
            expires={150}
          >
            Stilbruch.dev nutzt Cookies um die Benutzerfreundlichkeit zu
            verbessern.&nbsp;
            <Link
              style={{
                fontSize: "1.5vw",
                textDecoration: "underline",
                color: "rgb(255, 0, 0)"
              }}
              to="/datenschutz"
            >
              Infos zum Datenschutz
            </Link>
          </CookieConsent>
        </div>
      </Router>
    </Provider>
  );
}
