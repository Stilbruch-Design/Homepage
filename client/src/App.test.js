import React from "react";
import Enzyme, { shallow } from "enzyme";
import EnzymeAdapter from "enzyme-adapter-react-16";
import App from "./App";

Enzyme.configure({ adapter: new EnzymeAdapter() });

const setup = (props = {}, state = null) => {
  return shallow(<App {...props} />);
};

const findByTestAttr = (wrapper, val) => {
  return wrapper.find(`[data-test="${val}"]`);
};

test("renders without crashing", () => {
  const wrapper = setup();
  const appComponent = findByTestAttr(wrapper, "App_Component");
  expect(appComponent.length).toBe(1);
});
