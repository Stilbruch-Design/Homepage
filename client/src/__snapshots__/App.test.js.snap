// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders 1`] = `
<div
  data-testid="AppComponent"
>
  <div
    class="sc-EHOje bByzag"
    data-testid="NavbarComponent"
  >
    <a
      class="sc-bdVaJa fxCile"
      data-testid="NavLogo"
      href="/#Start"
      style="min-width: 10%;"
    >
      <object
        class="sc-bwzfXH cIvbrm"
      >
        <svg>
          tLogo.svg
        </svg>
      </object>
    </a>
    <a
      class="sc-bdVaJa fxCile"
      href="/#Entwicklung"
    >
      Entwicklung
    </a>
    <a
      class="sc-bdVaJa fxCile"
      href="/#Projekte"
    >
      Projekte
    </a>
    <a
      class="sc-bdVaJa fxCile"
      href="/#Story"
    >
      Story
    </a>
    <a
      class="sc-bdVaJa fxCile"
      href="/#Kontakt"
    >
      Kontakt
    </a>
  </div>
  <div
    class="sc-hMqMXs cbZZfH"
    data-testid="MainComponent"
  >
    <div
      class="sc-gZMcBi cnNjIV"
      data-testid="LandingComponent"
      id="Start"
    >
      <object
        class="sc-iwsKbI eOwdKt"
      >
        <svg
          class="navChange"
        >
          stilbruchLogo.svg
        </svg>
      </object>
      <div
        class="navChange sc-gqjmRU cxqoMz"
      >
        Webentwicklung
      </div>
    </div>
    <div
      class="navChange sc-VigVT fgFvrU"
      id="Entwicklung"
    >
      <div
        class="sc-jTzLTM hTYmZX"
      >
        Entwicklung
      </div>
      <div
        class="sc-jzJRlG ldJapp"
      >
        <div
          class="sc-fjdhpX iquxwt"
        >
          Javascript
        </div>
        <div
          class="sc-fjdhpX iquxwt"
        >
          HTML 5
        </div>
        <div
          class="sc-fjdhpX iquxwt"
        >
          CSS
        </div>
        <div
          class="sc-fjdhpX iquxwt"
        >
          Node
        </div>
        <div
          class="sc-fjdhpX iquxwt"
        >
          Express
        </div>
        <div
          class="sc-fjdhpX iquxwt"
        >
          React
        </div>
        <div
          class="sc-fjdhpX iquxwt"
        >
          Styled-Components
        </div>
        <div
          class="sc-fjdhpX iquxwt"
        >
          Redux
        </div>
        <div
          class="sc-fjdhpX iquxwt"
        >
          MongoDB
        </div>
        <div
          class="sc-fjdhpX iquxwt"
        >
          Jest
        </div>
        <div
          class="sc-fjdhpX iquxwt"
        >
          React Native
        </div>
        <div
          class="sc-fjdhpX iquxwt"
        >
          VS Code
        </div>
      </div>
    </div>
    <div
      class="navChange sc-VigVT fgFvrU"
      id="Projekte"
    >
      <div
        class="sc-jTzLTM hTYmZX"
      >
        Projekte
      </div>
      <div
        class="sc-cSHVUG hiTCKe"
      >
        <a
          class="sc-kAzzGY kLuLXp"
          href="https://www.radblocs.com"
        >
          <img
            alt="logo"
            src="logo_radblocs.png"
            style="max-width: 90%;"
          />
        </a>
      </div>
    </div>
    <div
      class="myFlex sGrid navChange sc-VigVT fgFvrU"
      id="Story"
    >
      <div
        class="sc-jTzLTM hTYmZX"
      >
        Story
      </div>
      <div
        class="sc-cSHVUG hiTCKe"
      >
        <span
          class="sc-kgoBCf fZlJqU"
        >
          Als selbst erlernter Webentwickler versuche ich den Problemen meines ärztlichen Alltages mit Hilfe von Programmcode und der Entwicklung intelligenter Software-Tools zu entgegnen. Funktionalität und Zuverlässigkeit sind mir dabei wichtiger als Komplexität und Design. Meine Webapplikationen reduzieren und erleichtern konventionelle und repetitive Arbeitsprozesse in der Medizin. Das spart Zeit, verbessert die Patientenversorgung und macht einfach mehr Spaß. Die Verbindung konservativer Medizin mit den vielfältigen Möglichkeiten moderner Webentwicklung ist der 
          <object
            class="sc-chPdSV fYNHqA"
          >
            <svg>
              stilbruchLogoShort.svg
            </svg>
          </object>
           dem ich mich widme.
          <span
            class="sc-kGXeez cEmXls"
          >
            -- Dr. M. Hübner
          </span>
        </span>
      </div>
    </div>
    <div
      class="navChange sc-VigVT fgFvrU"
      id="Kontakt"
    >
      <div
        class="sc-jTzLTM hTYmZX"
      >
        Kontaktform
      </div>
      <div
        class="sc-cSHVUG hiTCKe"
      >
        <form
          action="/kontakt"
          class="navChange"
          data-testid="FormMain"
          id="kontakt"
          method="POST"
        >
          <input
            class="sc-dxgOiQ dWWXhB"
            name="firstName1"
            style="display: none;"
            type="text"
            value=""
          />
          <input
            class="sc-dxgOiQ dWWXhB"
            name="lastName1"
            style="display: none;"
            type="text"
            value=""
          />
          <input
            class="sc-dxgOiQ dWWXhB"
            data-testid="InputEmail1"
            name="email1"
            style="display: none;"
            type="email"
            value=""
          />
          <textarea
            class="sc-ckVGcZ fPnDFe"
            name="message1"
            style="display: none;"
          />
          <input
            class="sc-dxgOiQ dWWXhB"
            data-testid="InputFirstName"
            minlength="2"
            name="firstName"
            placeholder="Vorname"
            required=""
            type="text"
            value=""
          />
          <input
            class="sc-dxgOiQ dWWXhB"
            data-testid="InputLastName"
            minlength="2"
            name="lastName"
            placeholder="Nachname"
            required=""
            type="text"
            value=""
          />
          <input
            class="sc-dxgOiQ dWWXhB"
            data-testid="InputEmail"
            name="email"
            placeholder="Email"
            required=""
            type="email"
            value=""
          />
          <textarea
            class="sc-ckVGcZ fPnDFe"
            data-testid="TextareaMessage"
            name="message"
            placeholder="Bitte schreibe hier deine Nachricht."
            required=""
          />
          <button
            class="sc-kpOJdX kiuHdg"
            data-testid="ButtonSubmit"
            type="submit"
          >
            <span>
              Nachricht senden
            </span>
          </button>
        </form>
      </div>
    </div>
  </div>
  <hr />
  <div
    class="sc-dnqmqq gOdgDC"
    data-testid="FooterMain"
  >
    <a
      class="sc-gzVnrw hwvJIi"
      data-testid="FooterLogo"
      href="/#Start"
    >
      <object
        class="sc-bZQynM fxTpvg"
      >
        <svg>
          tLogo.svg
        </svg>
      </object>
    </a>
    <a
      class="sc-htoDjs ytqdN"
      data-testid="FooterItemKontakt"
      href="/#Kontakt"
    >
      │ Kontakt 
    </a>
    <a
      class="sc-htoDjs ytqdN"
      data-testid="FooterItemDatenschutz"
      href="/datenschutz/#top"
    >
      │ Datenschutz 
    </a>
    <a
      class="sc-htoDjs ytqdN"
      data-testid="FooterItemImpressum"
      href="/impressum/#top"
    >
      │ Impressum
    </a>
  </div>
  <div
    class="sc-hSdWYo cNIshu"
  >
    <div
      class="cookieConsent "
      style="align-items: center; background: rgba(24, 24, 24, 0.9); color: rgb(255, 0, 0); display: flex; flex-wrap: wrap; justify-content: space-around; left: 0px; position: fixed; width: 100%; z-index: 999; flex-direction: row; border-radius: 10px; box-shadow: 5px 10px 20px rgba(0,0,0, 0.8); text-align: center; text-decoration: none; font-size: 2rem; font-weight: bold; margin-bottom: 1rem; bottom: 0px;"
    >
      <div
        class=""
        style="margin: 15px;"
      >
        Stilbruch.dev nutzt Cookies um die Benutzerfreundlichkeit zu verbessern. 
        <a
          href="/datenschutz"
          style="font-size: 1.5vw; text-decoration: underline; color: rgb(255, 0, 0);"
        >
          Infos zum Datenschutz
        </a>
      </div>
      <button
        class=""
        id=""
        style="background: rgba(95, 95, 95, 0.3); border: 1px solid #ff0000; border-radius: 10px; box-shadow: 5px 10px 20px rgba(0,0,0, 0.5); color: rgb(255, 0, 0); cursor: pointer; padding: 5px 10px; margin: 15px; font-size: 2rem; font-family: inherit; display: flex; flex-direction: column; justify-content: center; outline: none; font-weight: bold;"
      >
        Ok, stimme zu.
      </button>
    </div>
  </div>
</div>
`;
